crates/ink/codegen/src/generator/dispatch.rs

```diff
@@ -607,28 +624,32 @@ impl Dispatch<'_> {
                     let result: #constructor_output = #constructor_callable(input);
                     let output_value = ::ink::reflect::ConstructorOutputValue::new(result);
 
-                    match #constructor_value :: as_result(&output_value) {
+                    match #constructor_value::as_result(&output_value) {
                         ::core::result::Result::Ok(contract) => {
                             ::ink::env::set_contract_storage::<::ink::primitives::Key, #storage_ident>(
                                 &<#storage_ident as ::ink::storage::traits::StorageKey>::KEY,
                                 contract,
                             );
+
                             // only fallible constructors return success `Ok` back to the caller.
-                            if #constructor_value :: IS_RESULT {
-                                ::ink::env::return_value::<::core::result::Result<&(), ()>>(
+                            if #constructor_value::IS_RESULT {
+                                ::ink::env::return_value::<::ink::ConstructorResult<::core::result::Result<(), ()>>>(
```

Comment:
Doesn't really matter because how SCALE works but I think this helps with readability (at least I stumbled over it).

```suggestion
                                ::ink::env::return_value::<::ink::ConstructorResult<::core::result::Result<(), &#constructor_value::Error>>>(
```